IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -19,207 +19,273 @@ import AWSMobileClient
+        // No certificate ID has been stored in the user defaults; check to see if any .p12 files
+        // exist in the bundle.
+        let myBundle = Bundle.main
+        let myImages = myBundle.paths(forResourcesOfType: "p12" as String, inDirectory:nil)
+        let uuid = UUID().uuidString
+        
+        if (myImages.count > 0) {
+            // At least one PKCS12 file exists in the bundle.  Attempt to load the first one
+            // into the keychain (the others are ignored), and set the certificate ID in the
+            // user defaults as the filename.  If the PKCS12 file requires a passphrase,
+            // you'll need to provide that here; this code is written to expect that the
+            // PKCS12 file will not have a passphrase.
+            if let data = try? Data(contentsOf: URL(fileURLWithPath: myImages[0])) {
+                DispatchQueue.main.async {
+                    self.logTextView.text = "found identity \(myImages[0]), importing..."
+                }
+                if AWSIoTManager.importIdentity( fromPKCS12Data: data, passPhrase:"", certificateId:myImages[0]) {
+                    // Set the certificate ID and ARN values to indicate that we have imported
+                    // our identity from the PKCS12 file in the bundle.
+                    defaults.set(myImages[0], forKey:"certificateId")
+                    defaults.set("from-bundle", forKey:"certificateArn")
+                    DispatchQueue.main.async {
+                        self.logTextView.text = "Using certificate: \(myImages[0]))"
+                        self.iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:myImages[0], statusCallback: self.mqttEventCallback)
+                    }
+                }
             }
         }
-        else
-        {
-            activityIndicatorView.startAnimating()
-            logTextView.text = "Disconnecting..."
+        let certificateId = defaults.string( forKey: "certificateId")
+        return certificateId
+    }
 
-            DispatchQueue.global(qos: DispatchQoS.QoSClass.default).async {
-                self.iotDataManager.disconnect();
+    func createCertificateIdAndStoreinNSUserDefaults(onSuccess:  @escaping (String)->Void,
+                                                     onFailure: @escaping (Error) -> Void) {
+        let defaults = UserDefaults.standard
+        // Now create and store the certificate ID in NSUserDefaults
+        let csrDictionary = [ "commonName": CertificateSigningRequestCommonName,
+                              "countryName": CertificateSigningRequestCountryName,
+                              "organizationName": CertificateSigningRequestOrganizationName,
+                              "organizationalUnitName": CertificateSigningRequestOrganizationalUnitName]
+        
+        self.iotManager.createKeysAndCertificate(fromCsr: csrDictionary, callback: {  (responseOptional) -> Void in
```

Comment:
Prefer trailing closure syntax to named `callback` arg

Reply:
Updated.